### Disclaimer : 
##### Data direcly taken from Marvel.com Character Index which includes "team" names and are not technically superhero names. Its hard to manually check and remove them all, so I am ignoring such inconsistencies

### Import Names 

In [42]:
import pandas as pd
# Read the CSV file into a pandas data frame (df)
heros = pd.read_csv('rawHeroNames.csv', delimiter=',' ,header=None)

### Dataset Size

In [43]:
#number of super heros
heros.size

1491

### Lets have a look at few heros

In [44]:
heros.sample(10)

0
1080             Scarecrow (Ebenezer Laughton)
807                           Mikhail Rasputin
13                              Agent X (Nijo)
393                              Felicia Hardy
646                                 King Cobra
562                            Invisible Woman
158   Bride of Nine Spiders (Immortal Weapons)
211                              Carol Danvers
281                                  Daredevil
997                    Quasar (Wendell Vaughn)

### oops! Gotta get rid of those extra info in brackets and strip trailing spaces that are left

In [45]:
heros=heros[0].str.split("(",expand=True)
#strip space XXX" ""(YYY)-->XXX and not XXX" "
heros=pd.DataFrame(heros[0])
heros=heros[0].str.strip()
heros=pd.DataFrame(heros)

In [46]:
heros.sample(10)

0
1486               Zemo
1188         Spider-Man
867            Mystique
188     Captain America
479           Half-Life
728          Madame Web
1099             Sentry
1376            Vampiro
196      Captain Marvel
414   Franklin Richards

### Fixed? Not yet, Unfortunately we have more than 1 actors for each Hero. So, lets remove dups

In [47]:
print "before :" ,heros.size
heros = heros.drop_duplicates()
print "after :" , heros.size

before : 1491
after : 1257


### What to do about special chars, Caps, Spaces? We can take care of them during training 
### We have ara"ñ"a which when read as chars we get \xb1 \xc3 so lets rename it now

In [48]:
print heros[0][58]
heros[0][58]='Arana'
print heros[0][58]

Araña
Arana


In [49]:
heros.to_csv("SuperHeroCorpus.txt",sep='\n', encoding='utf-8', index=False, header=None)

### We now have a dataset of 1257 names.... It would be cool if we could visulaize the words and their 

### similarity, patterns? 

### Let's create our own Super Hero!